### PR TITLE
#490: Addition of instance_id in ListClusters api

### PR DESCRIPTION
**Pull request by hemant-qlogic**
Added Instance_id to the api ListClusters.
Some of the test cases are not run on the emulator.
Please let me know how to test these test cases, e.g. example is not running with emulator

### CODE REVIEW

File: bigtable/client/instance_admin.cc [hunk 1]
```diff
@@ -125,9 +125,10 @@ void InstanceAdmin::DeleteInstance(std::string const& instance_id) {
   }
 }
 
-std::vector<btproto::Cluster> InstanceAdmin::ListClusters() {
+std::vector<btproto::Cluster> InstanceAdmin::ListClusters(
```
Comment (coryan) [inline bigtable/client/instance_admin.cc]: Please add an overload without an `instance_id` parameter that calls the normal implementation, as follows:

```C++
std::vector<btproto::Cluster> InstanceAdmin::ListClusters() {
  return ListClusters("-");
}
```

According to the proto documentation `-` is the magic instance that means "any instance".  We want to support listing all clusters, but do not want the users to know about the magic strings.

File: bigtable/client/instance_admin.h [hunk 2]
```diff
@@ -56,6 +56,18 @@ class InstanceAdmin {
...
   std::string const& project_id() const { return impl_.project_id(); }
 
+  /// Return the fully qualified name of the given instance_id.
+  std::string InstanceName(std::string const& instance_id) const {
+    return impl_.InstanceName(instance_id);
+  }
+
+  /// Return the fully qualified name of the given cluster_id of the given
+  /// instance_id.
+  std::string ClusterName(bigtable::InstanceId const& instance_id,
```
Comment (coryan) [inline bigtable/client/instance_admin.h]: I do not think `ClusterName()` is used in this PR, please do not include it.

File: bigtable/examples/bigtable_samples_instance_admin.cc [hunk 3]
```diff
@@ -117,7 +117,11 @@ void DeleteInstance(bigtable::InstanceAdmin instance_admin, int argc,
 //! [list clusters]
 void ListClusters(bigtable::InstanceAdmin instance_admin, int argc,
                   char* argv[]) {
-  auto cluster_list = instance_admin.ListClusters();
+  if (argc != 3) {
+    throw Usage{"list-clusters <project-id> <instance-id> <cluster-id>"};
```
Comment (coryan) [inline bigtable/examples/bigtable_samples_instance_admin.cc]: Please fix usage string, `<cluster-id>` is not required.

File: bigtable/tests/instance_admin_integration_test.cc [hunk 4]
```diff
@@ -123,13 +123,30 @@ TEST_F(InstanceAdminIntegrationTest, ListClustersTest) {
...
+          .set_type(bigtable::InstanceConfig::DEVELOPMENT);
+  auto instance_details =
+      instance_admin_->CreateInstance(instance_config).get();
+
+  // Create clusters in an instance
+  // TODO(#422) - Implement InstanceAdmin::CreateCluster
 
   // TODO(#418) - create an instance and test that its cluster is returned here.
-  auto clusters = instance_admin_->ListClusters();
+  auto clusters = instance_admin_->ListClusters(id);
```
Comment (coryan) [inline bigtable/tests/instance_admin_integration_test.cc]: This test would pass even if `ListClusters()` returns an empty vector.  Please check that the collection is not empty.

**Comment by hemant-qlogic** [comment]
PTAL

File: firestore/client/firestore_client.bzl [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# DO NOT EDIT -- GENERATED CODE
```
Comment (coryan) [inline firestore/client/firestore_client.bzl]: Please remove this file from this PR.

File: firestore/client/firestore_client_testing.bzl [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# DO NOT EDIT -- GENERATED CODE
```
Comment (coryan) [inline firestore/client/firestore_client_testing.bzl]: Please remove this file from this PR.

File: firestore/client/firestore_client_unit_tests.bzl [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# DO NOT EDIT -- GENERATED CODE
```
Comment (coryan) [inline firestore/client/firestore_client_unit_tests.bzl]: Please remove this file from this PR.

File: bigtable/examples/bigtable_samples_instance_admin.cc [hunk 8]
```diff
@@ -117,14 +117,33 @@ void DeleteInstance(bigtable::InstanceAdmin instance_admin, int argc,
...
+
+  auto cluster_list = instance_admin.ListClusters(ConsumeArg(argc, argv));
   std::cout << "Cluster Name List" << std::endl;
   for (auto const& cluster : cluster_list) {
     std::cout << "Cluster Name:" << cluster.name() << std::endl;
   }
 }
 //! [list clusters]
 
+//! [list all clusters]
```
Comment (coryan) [inline bigtable/examples/bigtable_samples_instance_admin.cc]: In a future PR we need to implement the `ListClusters` function in the instance admin emulator and then we need to call these new examples for `run_examples_utils.sh`.  Can you make a note in #490 to that effect or create a new bug for it?

File: firestore/client/firestore_client.bzl (hunk 5 above)
Comment (hemant-qlogic) [reply firestore/client/firestore_client.bzl]: I did not get why to create separate PR for the bzl files.
What should be the title for the new PR for bzl files.

Comment (hemant-qlogic) [reply firestore/client/firestore_client.bzl]: All Build looking as successful in last commit, which build was broken.

**Comment by hemant-qlogic** [comment]
related issue #528  :Implement ListClusters in instance admin emulator

**Comment by hemant-qlogic** [comment]
Some builds fails due to missing bzl files.
====================Error in build===================
Loading: 
Loading: 0 packages loaded
Loading: 0 packages loaded
    currently loading: firestore/client
ERROR: error loading package 'firestore/client': Extension file not found. Unable to load file '//firestore/client:firestore_client_unit_tests.bzl': file doesn't exist or isn't a file
Loading: 0 packages loaded
    currently loading: firestore/client
ERROR: error loading package 'firestore/client': Extension file not found. Unable to load file '//firestore/client:firestore_client_unit_tests.bzl': file doesn't exist or isn't a file
Loading: 0 packages loaded
    currently loading: firestore/client
INFO: Elapsed time: 0.788s
====================Error in build===================
If I am not adding bzl files then builds fails, So how to add it into a separate PR.

**Comment by coryan** [comment]
The .bzl files were added as part of #519.  You probably need to rebase.

**Comment by coryan** [comment]
Are you planning to implement #528 first and then complete this PR?  Or would you like me to review as-is?

**Comment by hemant-qlogic** [comment]
I think if this gets reviewed first then I will have less changes to commit, otherwise I need to merge multiple changes of other PRS and will take more time for merging.
So if this gets completed I can start on #528

File: firestore/client/firestore_client.bzl [hunk 9]
```diff
@@ -1,9 +0,0 @@
-# DO NOT EDIT -- GENERATED BY CMake -- Change the CMakeLists.txt file if needed
```
Comment (coryan) [inline firestore/client/firestore_client.bzl]: ooops.  I asked you to remove this file before #519 got merged, now it is, please do not remove it.

Comment (hemant-qlogic) [reply firestore/client/firestore_client.bzl]: Shall I add these files in my PR.

Comment (coryan) [reply firestore/client/firestore_client.bzl]: Your goal should be to leave these files in the same state they are in the branch you are merging to.

If your branch was rebased to c3cbd8f67d92af3d351cf9989ac4b27e5aacf541 (or any newer commit on `master`), then the files are in `master` and you should not erase them, and you should have no need to modify them in any way.

If your branch was based before that commit then you should not add them.

**Comment by hemant-qlogic** [comment]
PTAL

**Pull request merged**